src/limitations/prerelease_versions.md

```diff
@@ -0,0 +1,64 @@
+# Pre-release versions
+
+Pre-releasing is a very common pattern in the world of versioning.
+It is however one of the worst to take into account in a dependency system, and I highly recommend that if you can avoid introducing pre-releases in your package manager, you should.
+In the context of pubgrub, pre-releases break two fondamental properties of the solver.
+
+1. Pre-releases act similar to continuous spaces.
+2. Pre-releases break the mathematical properties of subsets in a space with total order.
+
+(1) Indeed, it is hard to answer what version comes after "1-alpha0".
+Is it "1-alpha1", "1-beta0", "2"?
+In practice, we could say that the version that comes after "1-alpha0" is "1-alpha0?" where the "?" character is chosen to be the lowest character in the lexicographic order, but we clearly are on a stretch here and it certainly isn't natural.
+
+(2) Pre-releases are often semantically linked to version constraints written by humans, interpreted differently depending on context.
+For example, "2.0.0-beta" is meant to exist previous to version "2.0.0".
+Yet, it is not supposed to be contained in the set described by `1.0.0 <= v < 2.0.0`, and only within sets where one of the bounds contains a pre-release marker such as `2.0.0-alpha <= v < 2.0.0`.
+This poses a problem to the dependency solver because of backtracking.
+Indeed, the PubGrub algorithm relies on knowledge accumulated all allong the propagation of the solver front.
+And this knowledge is composed of facts, that are thus never backtracked even when backtracking happens.
```

Comment:
"thus never backtracked even when backtracking happens" -> "thus never removed even when backtracking happens" 
Just to reduce the use of "backtracking" in this paragraph.